### PR TITLE
ci/nightly: permit skipping tests when manually triggering build

### PR DESCRIPTION
**Pull request by benesch**
### Checklist

- [x] This PR has adequate test coverage / QA involvement has been duly considered.
- [x] This PR adds a release note for any [user-facing behavior changes](https://github.com/MaterializeInc/materialize/blob/main/doc/user/content/release-notes.md#what-changes-require-a-release-note).

### CODE REVIEW

**Comment by benesch** [comment]
Ok, @philip-stoev, I think this should do what you want! When you trigger a build from the UI now, you're prompted to select which tests you want to run: https://buildkite.com/materialize/nightlies/builds/657. Feel free to press "Rebuild" on that build to try it out for yourself.

The scheduled build will continue to run all tests automatically. (At least, I think; can't test that until this merges.)

File: ci/nightly/pipeline.yml [hunk 1]
```diff
@@ -7,115 +7,151 @@
...
+          - { value: cluster-testdrive }
+          - { value: proxy }
+          - { value: testdrive-workers-1 }
+          - { value: persistence-testdrive }
+          - { value: feature-benchmark }
+          - { value: feature-benchmark-persistence }
+          - { value: aws-config }
+        multiple: true
+        required: false
+    if: build.source == "ui"
```
Comment (philip-stoev) [inline ci/nightly/pipeline.yml:36]: So your new TV show is called "Yaml is the new bash" :-) The conditional is at the end of the block it references ...

**Review by philip-stoev — APPROVED** [review]
Thanks a lot ! Seems to work as advertised!

**Review by philip-stoev — APPROVED** [review]

**Comment by philip-stoev** [comment]
I am merging this now as I will be making changes to the nightly pipeline in a different PR shortly.

**Pull request merged**